python/vowpalwabbit/pyvw.py

```diff
@@ -1158,6 +1160,74 @@ def __str__(self):
         )
 
 
+class CCBLabelType(enum.Enum):
+    UNSET = pylibvw.vw.tUNSET
+    SHARED = pylibvw.vw.tSHARED
+    ACTION = pylibvw.vw.tACTION
+    SLOT = pylibvw.vw.tSLOT
+
+
+class CCBSlotOutcome:
+    cost: float
+    action_probs: List[Tuple[int, float]]
+
+    def __init__(self, cost=0, action_probs=[], **kwargs):
+        ex = kwargs.get("ex")
+        if isinstance(ex, example):
+            self.from_example(ex)
+        else:
+            self.cost = cost
+            self.action_probs = action_probs
+
+    def from_example(self, ex):
+        self.cost = ex.get_ccb_cost()
+        self.action_probs = []
+        for i in range(ex.get_ccb_num_included_actions()):
+            self.action_probs.append((ex.get_ccb_class(i), ex.get_ccb_probs(i)))
+
+    def __str__(self):
+        top_action, top_prob = self.action_probs[0]
+        out = f"{top_action}:{top_prob}:{self.cost}"
+        for action, prob in self.action_probs[1:]:
+            out += f",{action}:{prob}"
+        return out
+
+
+class CCBLabel(abstract_label):
+    """Class for conditional contextual bandits VW label"""
+
+    type: CCBLabelType
+    explicit_included_actions: List[int]
+    weight: float
+    outcome: Optional[CCBSlotOutcome]
+
+    def __init__(
+        self,
+        type=CCBLabelType.UNSET,
+        explicit_included_actions=[],
+        weight=1,
+        outcome=None,
+        **kwargs,
+    ):
+        abstract_label.__init__(self)
+        ex = kwargs.get("ex")
+        if isinstance(ex, example):
+            self.type = ex.get_ccb_type()
+            self.explict_included_actions = ex.get_ccb_explicitly_included_actions()
+            self.weight = ex.get_ccb_weight()
+            self.outcome = (
+                CCBSlotOutcome(ex=ex) if self.type == CCBLabelType.SLOT else None
+            )
+        else:
+            self.type = type
+            self.explicit_included_actions = explicit_included_actions
+            self.weight = weight
+            self.outcome = outcome
+
+    def __str__(self):
+        return f"ccb {self.type} {str(self.outcome or '')} {str(self.explicit_included_actions or '')}"
```

Comment:
`self.type` needs some sort of string conversion since it is an enum here

Reply:
`self.type.name.lower()` used this trick.